app/models/user.rb

```diff
@@ -174,4 +176,10 @@ def activate!
   def deactivate!
     update! active: false
   end
+
+  def check_user_role_provider
+    return unless role && role_changed?
```

Comment:
We can actually move this check up to the `validate` line. So `validate :check_user_role_provider if role_changed?`

Reply:
Nice